src/common/telemetry/src/metric.rs

```diff
@@ -25,3 +33,233 @@ pub fn dump_metrics() -> Result<String, String> {
+                MetricType::SUMMARY => {
+                    let s = m.get_summary();
+                    for q in s.get_quantile() {
+                        timeseries.push(TimeSeries {
+                            labels: convert_label(
+                                m.get_label(),
+                                mf_name,
+                                Some(("quantile", q.get_quantile().to_string())),
+                            ),
+                            samples: vec![Sample {
+                                value: q.get_value(),
+                                timestamp,
+                            }],
+                            exemplars: vec![],
+                        });
+                    }
+                    timeseries.push(TimeSeries {
+                        labels: convert_label(
+                            m.get_label(),
+                            format!("{}_sum", mf_name).as_str(),
+                            None,
+                        ),
+                        samples: vec![Sample {
+                            value: s.get_sample_sum(),
+                            timestamp,
+                        }],
+                        exemplars: vec![],
+                    });
+                    timeseries.push(TimeSeries {
+                        labels: convert_label(
+                            m.get_label(),
+                            format!("{}_count", mf_name).as_str(),
+                            None,
+                        ),
+                        samples: vec![Sample {
+                            value: s.get_sample_count() as f64,
+                            timestamp,
+                        }],
+                        exemplars: vec![],
+                    });
+                }
+                MetricType::UNTYPED => {
+                    // `TextEncoder` `MetricType::UNTYPED` unimplemented
+                    // To keep the implementation consistent and not cause unexpected panics, we do nothing here.
+                }
+            };
+        }
+    }
+    remote::WriteRequest {
+        timeseries,
+        metadata: vec![],
+    }
+}
+
+fn convert_label(
+    pairs: &[LabelPair],
+    name: &str,
+    addon: Option<(&'static str, String)>,
+) -> Vec<remote::Label> {
+    let mut labels = Vec::with_capacity(pairs.len() + 1 + if addon.is_some() { 1 } else { 0 });
+    for label in pairs {
+        labels.push(remote::Label {
+            name: label.get_name().to_string(),
+            value: label.get_value().to_string(),
+        });
+    }
+    labels.push(remote::Label {
+        name: "__name__".to_string(),
+        value: name.to_string(),
+    });
+    if let Some(addon) = addon {
+        labels.push(remote::Label {
+            name: addon.0.to_string(),
+            value: addon.1,
+        });
+    }
+    // Remote write protocol need label names sorted in lexicographical order.
+    labels.sort_unstable_by(|a, b| a.name.cmp(&b.name));
+    labels
+}
```

Comment:
Can you provide some unit tests to check these functions?

Reply:
Unit test has added